files/tools/terraform_env_variable_helper.sh

```diff
@@ -35,13 +35,13 @@ echo -e "\nTF_VAR_subscription_id:"
 echo -e "$TF_VAR_subscription_id"
 
 # obtain the client ID
-TF_VAR_client_id=$(az ad sp show --id http://"$YOURSP" --query appId --output tsv)
+TF_VAR_client_id=$(az ad sp list --display-name "$YOURSP" --query '[].appId' --output tsv)
 export TF_VAR_client_id
 echo -e "\nTF_VAR_client_id for this Service Principal $YOURSP:"
 echo -e "$TF_VAR_client_id"
 
 # obtain a client secret
-TF_VAR_client_secret=$(az ad sp create-for-rbac --skip-assignment --name http://"$YOURSP" --query password --output tsv)
+TF_VAR_client_secret=$(az ad sp create-for-rbac --skip-assignment --name "$YOURSP" --query password --output tsv)
```

Comment:
Do see where the old command does now set the name to include `http://` with your Service Principal name. Would be nice to know what/how this changed. If you can find docs and or info that would be helpful. We'll need to adjust other items in the `How-to` doc for this project as well when creating a service principal.

Reply:
According to the [documentation for the az ad sp module](https://docs.microsoft.com/en-us/cli/azure/ad/sp?view=azure-cli-latest#az_ad_sp_create_for_rbac), it seems to always use the latest version instead of specific versions, so after some searching, I was not able to pinpoint this to a specific change in the AZ CLI from version to version.  Perhaps, it would be possible to search through the[ GitHub project ](https://github.com/Azure/azure-cli) and find this, but I did not have time for that.

I just know that this new command works as expected with the version of the AZ CLI that I mentioned above for me as well as a colleague.  

If you would need, I could add a usage note or another echo in the script with the version number this was developed and validated using.

Reply:
No need. Just need to adjust our auth docs now.